spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/integration/IntegrationAutoConfiguration.java

```diff
@@ -110,6 +115,24 @@
 	@EnableIntegration
 	protected static class IntegrationConfiguration {
 
+		@Bean(PollerMetadata.DEFAULT_POLLER)
+		@ConditionalOnMissingBean(name = PollerMetadata.DEFAULT_POLLER)
+		public PollerMetadata defaultPoller(IntegrationProperties integrationProperties) {
+			IntegrationProperties.Poller poller = integrationProperties.getPoller();
+			PollerMetadata pollerMetadata = new PollerMetadata();
+			PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
+			map.from(poller::getMaxMessagesPerPoll).to(pollerMetadata::setMaxMessagesPerPoll);
+			map.from(poller::getReceiveTimeout).as(Duration::toMillis).to(pollerMetadata::setReceiveTimeout);
+			map.from(poller::getCron).whenHasText().as(CronTrigger::new).to(pollerMetadata::setTrigger);
+			map.from(poller.getFixedDelay() != null ? poller.getFixedDelay() : poller.getFixedRate())
+					.as(Duration::toMillis).as(PeriodicTrigger::new).as((trigger) -> {
+						map.from(poller::getInitialDelay).as(Duration::toMillis).to(trigger::setInitialDelay);
+						trigger.setFixedRate(poller.getFixedRate() != null);
+						return trigger;
+					}).to(pollerMetadata::setTrigger);
```

Comment:
I wonder how we should handle the user setting more than one of `cron`, `fixedDelay`, and `fixedRate`. As things stand it looks like the order of precedence is `fixedDelay`, `fixedRate`, `cron`. Should we document this ordering, fail when more than one is configured, or something else?

Reply:
We've decided to fail.